lib/at_cmd_parser/src/at_cmd_decoder.c

```diff
@@ -0,0 +1,277 @@
+#include <zephyr.h>
+
+#include <at_cmd_parser.h>
+#include <at_cmd_decoder.h>
+
+#include <string.h>
+
+/* Forward declarations. */
+static void *at_CEREG_decode(struct at_param_list *param_list,
+			     u32_t valid_params);
+
+static void *at_CESQ_decode(struct at_param_list *param_list,
+			    u32_t valid_params);
+
+static void *at_NOT_CESQ_decode(struct at_param_list *param_list,
+				u32_t valid_params);
+
+static void *at_CMT_decode(struct at_param_list *param_list,
+			   u32_t valid_params);
+
+static void *at_CNUM_decode(struct at_param_list *param_list,
+			    u32_t valid_params);
+
+AT_CMD_DECODER_LIST_START(default_decoder_list)
+AT_CMD_DECODER_LIST_ENTRY("+CESQ", AT_CMD_CESQ_MODEL, at_CESQ_decode)
+AT_CMD_DECODER_LIST_ENTRY("%CESQ", AT_NOT_CESQ_MODEL, at_NOT_CESQ_decode)
```

Comment:
Yes, I was a bit back and forth with the naming scheme. I ended up on NOT (for notification) because it aligns with CMD. But I agree it's bad, NOT is to strong a word and can not be used for anything else. I should pobablt change this.